src/common/transformations/src/transformations/common_optimizations/mark_subgraphs_to_keep_in_mixed_precision.cpp

```diff
@@ -0,0 +1,363 @@
+        auto max_or_add = pattern::wrap_type<opset10::Maximum, opset10::Add>(OutputVector{input_2, eps_const_pattern});
+
+        auto sqrt = std::make_shared<opset10::Sqrt>(max_or_add);
+        auto sqrt_or_max_add = std::make_shared<pattern::op::Or>(OutputVector{max_or_add, sqrt});
+        // whether is divided directly or after sqrt (e.g. in L2Norm after sqrt, in MVN is divided directly)
+        auto divide = std::make_shared<opset10::Divide>(input_1, sqrt_or_max_add);
+
+        auto pow_exp = pattern::wrap_type<opset10::Constant>();
+        auto convert_pattern = pattern::wrap_type<opset10::Convert>({pow_exp});
+        auto pow_exp_or_convert = std::make_shared<pattern::op::Or>(OutputVector{pow_exp, convert_pattern});
+
+        auto pow_pattern = std::make_shared<opset10::Power>(max_or_add, pow_exp_or_convert);
+        auto mul_pattern = std::make_shared<opset10::Multiply>(input_1, pow_pattern);
+        auto div_or_mul_to_negative_pow = std::make_shared<pattern::op::Or>(OutputVector{divide, mul_pattern});
+
+        matcher_pass_callback callback = [=](pattern::Matcher& m) {
+            const auto& pattern_to_output = m.get_pattern_map();
+            if (!m.get_match_root())
+                return false;
+
+            const auto mul = std::dynamic_pointer_cast<opset10::Multiply>(m.get_match_root());
+            // if pattern input_1*Pow(Maximum(input_2, eps), z) or input_1*Pow(Add(input_2, eps), z) is matched
+            // need to check that power is negative
+            if (mul) {
+                const auto pow_const = std::dynamic_pointer_cast<opset10::Constant>(pattern_to_output.at(pow_exp));
+                if (pow_const) {
+                    // continue only if exponent is negative (z < 0)
+                    if (pow_const->get_element_type() == element::f16) {
+                        for (auto val : pow_const->get_vector<float16>())
+                            if (val >= 0.0f)
+                                return false;
+                    } else if (pow_const->get_element_type() == element::f32) {
+                        for (auto val : pow_const->get_vector<float>())
+                            if (val >= 0.0f)
+                                return false;
+                    }
+                }
+            }
+
+            const auto eps_const =
+                std::dynamic_pointer_cast<opset10::Constant>(pattern_to_output.at(eps_const_pattern));
+            if (!eps_const)
+                return false;
+            if (eps_const->get_element_type() == element::f32) {
+                for (const auto& val : eps_const->get_vector<float>())
+                    if (val > static_cast<float>(float16::from_bits(0x0400)))
+                        return false;
+            } else if (eps_const->get_element_type() == element::f16) {
+                for (const auto& val : eps_const->get_vector<float16>())
+                    if (val > float16::from_bits(0x0400))
```

Comment:
met this constant `float16::from_bits(0x0400)` twice. May be to create this constant in `mark_subgraphs_to_keep_in_mixed_precision.hpp`?

Reply:
defined a new const `float16_min_normalized` in hpp file